uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/models/leave-request.model.js

```diff
@@ -30,18 +30,11 @@ define([
        * @return {Instance} instance of the request depending on its type
        */
       function initRequestInstance (requestObject) {
-        var requestInstance;
-        var requestType = requestObject.request_type;
-
-        if (requestType === 'sickness') {
-          requestInstance = SicknessRequestInstance;
-        } else if (requestType === 'toil') {
-          requestInstance = TOILRequestInstance;
-        } else {
-          requestInstance = LeaveRequestInstance;
-        }
-
-        return requestInstance.init(requestObject, true);
+        return ({
```

Comment:
not sure if you saw [this comment](https://github.com/civicrm/civihr/pull/2361#discussion_r157799633)

Reply:
I am not sure we would like to call $injector.get within every leave request fetched via API especially via `all()`.